Houseclub/src/main/java/me/grishka/houseclub/fragments/ProfileFragment.java

```diff
@@ -288,9 +303,98 @@ public void onError(ErrorResponse error){
 
 	private void onInstagramClick(View v){
+		if (self){
+			HashMap<String, String> headers = new HashMap<>();
+			headers.put("CH-AppBuild", ClubhouseAPIController.API_BUILD_ID);
+			headers.put("CH-AppVersion", ClubhouseAPIController.API_BUILD_VERSION);
+			headers.put("User-Agent", ClubhouseAPIController.API_UA);
+
+			headers.put("CH-DeviceId", ClubhouseSession.deviceID);
+			headers.put("Authorization", "Token "+ClubhouseSession.userToken);
+			headers.put("CH-UserID", ClubhouseSession.userID);
+
+			if (user.instagram == null) {
+				webView.setVisibility(View.VISIBLE);
+				webView.setWebViewClient(new WebViewClient() {
+					@Override
+					public boolean shouldOverrideUrlLoading(WebView view, WebResourceRequest request) {
+						Boolean redirect = checkRedirect(request.getUrl().toString());
+						view.loadUrl(request.getUrl().toString());
+						return redirect;
+					}
+
+					public boolean shouldOverrideUrlLoading(WebView view, String url) {
+						Boolean redirect = checkRedirect(url);
+						view.loadUrl(url);
+						return redirect;
+					}
+				});
+				webView.loadUrl(
+						"https://www.instagram.com/oauth/authorize?client_id=" +
+								BuildConfig.INSTAGRAM_APP_ID +
+								"&redirect_uri=" + UpdateInstagram.REDIRECT_INSTAGRAM_URL +
+								"&scope=user_profile" +
+								"&response_type=code",
+						headers
+				);
+			} else {
+				new AlertDialog.Builder(getActivity())
+						.setMessage(getString(R.string.confirm_unlink_instagram_title))
+						.setMessage(getString(R.string.confirm_unlink_instagram))
+						.setPositiveButton(R.string.yes, new DialogInterface.OnClickListener(){
+							@Override
+							public void onClick(DialogInterface dialogInterface, int i){
+								new UpdateInstagram(null)
+										.wrapProgress(getActivity())
+										.setCallback(new Callback<BaseResponse>(){
+											@Override
+											public void onSuccess(BaseResponse result){
+												instagram.setText(R.string.add_instagram);
+												webView.setVisibility(View.GONE);
+											}
+
+											@Override
+											public void onError(ErrorResponse error){
+												error.showToast(getActivity());
+											}
+										})
+										.exec();
+							}
+						})
+						.setNegativeButton(R.string.no, null)
+						.show();
+			}
+		} else
 		startActivity(new Intent(Intent.ACTION_VIEW, Uri.parse("https://instagram.com/"+user.instagram)));
 	}
 
+	private Boolean checkRedirect(String url){
+		if (url.startsWith(UpdateInstagram.REDIRECT_INSTAGRAM_URL)) {
+
+			// last2 chars is #_ by docs https://developers.facebook.com/docs/instagram-basic-display-api/getting-started
+			String code = url.substring((UpdateInstagram.REDIRECT_INSTAGRAM_URL+ "?code=").length(), url.length()-2);
+
+			new UpdateInstagram(code)
+					.wrapProgress(getActivity())
+					.setCallback(new Callback<BaseResponse>(){
+						@Override
+						public void onSuccess(BaseResponse result){
+							instagram.setText(R.string.instagram_linked);
```

Comment:
add `user.instagram = null;`

Reply:
it is incorrect. better is make request to update user (in all type of change situation)

Reply:
Plz can you give me your insta have alot of questions to ask you about

Reply:
@zerezhka but it causes a bug, look what happens when you click after unlinking

Reply:
i'll see, thx